src/channel.ts

```diff
@@ -51,19 +51,22 @@ export const OPACITY = Channel.OPACITY;
 export const CHANNELS = [X, Y, X2, Y2, ROW, COLUMN, SIZE, SHAPE, COLOR, ORDER, OPACITY, TEXT, DETAIL];
 
 // CHANNELS without COLUMN, ROW
-export const UNIT_CHANNELS = [X, Y, X2, Y2, SIZE, SHAPE, COLOR, ORDER, OPACITY, TEXT, DETAIL];
+export const UNIT_CHANNELS = without(CHANNELS, [ROW, COLUMN] as Channel[]);
 
 // UNIT_CHANNELS without X2, Y2, ORDER, DETAIL, TEXT
-export const UNIT_SCALE_CHANNELS = [X, Y, SIZE, SHAPE, COLOR, OPACITY];
+export const UNIT_SCALE_CHANNELS = without(UNIT_CHANNELS, [X2, Y2, ORDER, DETAIL, TEXT] as Channel[]);
 
 // UNIT_SCALE_CHANNELS with ROW, COLUMN
-export const SCALE_CHANNELS = [X, Y, SIZE, SHAPE, COLOR, OPACITY, ROW, COLUMN];
+export const SCALE_CHANNELS = union(UNIT_SCALE_CHANNELS, [ROW, COLUMN] as Channel[]);
 
 // UNIT_CHANNELS without X, Y, X2, Y2;
-export const NONSPATIAL_CHANNELS = [SIZE, SHAPE, COLOR, ORDER, OPACITY, TEXT, DETAIL];
+export const NONSPATIAL_CHANNELS = without(UNIT_CHANNELS, [X, Y, X2, Y2] as Channel[]);
+
+// NONSPATIAL_CHANNELS_EXCEPT_ORDER without ORDER;
```

Comment:
1) This comment seems unnecessary
2) I'm not sure if we should move `NONSPATIAL_CHANNELS_EXCEPT_ORDER` here.  (It's only use once in `compile/mark/mark.ts`